src/rgw/rgw_dedup_utils.h

```diff
@@ -0,0 +1,436 @@
+#pragma once
+#include <string>
+#include "include/rados/buffer.h"
+#include "include/encoding.h"
+#include "common/Formatter.h"
+#include "common/ceph_json.h"
+#include <time.h>
+#include "include/utime.h"
+#include "include/encoding.h"
+
+//#define FULL_DEDUP_SUPPORT
+namespace rgw::dedup {
+  static constexpr const char* DEDUP_WATCH_OBJ = "DEDUP_WATCH_OBJ";
+  using work_shard_t   = uint16_t;
+  using md5_shard_t    = uint16_t;
+
+  // settings to help debug small systems
+  const work_shard_t MIN_WORK_SHARD = 4;
+  const md5_shard_t  MIN_MD5_SHARD  = 4;
+
+  // Those are the correct values for production system
+  const work_shard_t MAX_WORK_SHARD = 255;
+  const md5_shard_t  MAX_MD5_SHARD  = 512;
+
+  const work_shard_t NULL_WORK_SHARD = 0xFFFF;
+  const md5_shard_t  NULL_MD5_SHARD  = 0xFFFF;
+  const unsigned     NULL_SHARD      = 0xFFFF;
+
+  // work_shard  is an 8 bits int with 255 legal values for the first iteration
+  // and one value (0xFF) reserved for second iteration
+  const unsigned     WORK_SHARD_HARD_LIMIT = 0x0FF;
+  // md5_shard_t is a 12 bits int with 4096 possible values
+  const unsigned     MD5_SHARD_HARD_LIMIT  = 0xFFF;
+
+  static_assert(MAX_WORK_SHARD < NULL_WORK_SHARD);
+  static_assert(MAX_WORK_SHARD < NULL_SHARD);
+  static_assert(MAX_WORK_SHARD <= WORK_SHARD_HARD_LIMIT);
+  static_assert(MAX_MD5_SHARD  < NULL_MD5_SHARD);
+  static_assert(MAX_MD5_SHARD  < NULL_SHARD);
+  static_assert(MAX_MD5_SHARD  <= MD5_SHARD_HARD_LIMIT);
+
+  //---------------------------------------------------------------------------
+  enum dedup_req_type_t {
+    DEDUP_TYPE_NONE     = 0,
+    DEDUP_TYPE_ESTIMATE = 1,
```

Comment:
the radosgw admin code references `DEDUP_TYPE_DRY_RUN` ?

Reply:
I forgot to commit my changes to radosgw admin